aspnetcore/testing/razor-pages-testing/sample_snapshot/tests/RazorPagesTestingSample.Tests/UnitTests/DataAccessLayerTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+using RazorPagesTestingSample.Data;
+
+namespace RazorPagesTestingSample.Tests.UnitTests
+{
+    public class DataAccessLayerTests
+    {
+        [Fact]
+        public async Task GetMessagesAsync_MessagesAreReturned()
+        {
+            using (var db = new AppDbContext(Utilities.TestingDbContextOptions()))
+            {
+                // Arrange
+                var expectedMessages = AppDbContext.GetSeedingMessages();
+                await db.AddRangeAsync(expectedMessages);
+                await db.SaveChangesAsync();
+
+                // Act
+                var result = await db.GetMessagesAsync();
+
+                // Assert
+                var actualMessages = Assert.IsAssignableFrom<List<Message>>(result);
+                Assert.Equal(
+                    expectedMessages.OrderBy(x => x.Id), 
+                    actualMessages.OrderBy(x => x.Id), 
+                    new Utilities.MessageComparer());
```

Comment:
Could we change this to
```
Assert.Equal(
Assert.Equal(
    expectedMessages.OrderBy(x => x.Id).Select(m => m.Message), 
    actualMessages.OrderBy(x => x.Id).Select(m => m.Message));
```

and drop the comparer. The comparer type seems superfluous.

Reply:
Yup, this looks good.